internal/common/cloudproviderapi/client.go

```diff
@@ -80,6 +80,31 @@ func (c *Client) DeleteAWSAccount(ctx context.Context, stackID string, accountID
 	return nil
 }
 
+type AWSCloudWatchScrapeJob struct {
+	StackID               string
+	Name                  string
+	Enabled               bool
+	AWSAccountResourceID  string
+	Regions               []string
+	ServiceConfigurations []AWSCloudWatchServiceConfiguration
+}
+type AWSCloudWatchServiceConfiguration struct {
+	Name                        string
+	Metrics                     []AWSCloudWatchMetric
+	ScrapeIntervalSeconds       int64
+	ResourceDiscoveryTagFilters []AWSCloudWatchTagFilter
+	TagsToAddToMetrics          []string
+	IsCustomNamespace           bool
```

Comment:
Agreed on that point, good to know that support doesn't carry over, and makes sense. We might be best off implementing a separate resource + data sources for custom namespace jobs, then? Another idea could be to automatically designate the service name as a custom namespace on the backend if it doesn't match anything in our service maps, but that could be a bit of a surprising behavior as well.

Reply:
> Another idea could be to automatically designate the service name as a custom namespace on the backend if it doesn't match anything in our service maps, but that could be a bit of a surprising behavior as well.

The fun part with that is some customers have realized that they can setup a custom namespace for a service we don't have defined quite yet. It would be fun if in one terraform run it's a custom and the next it's not 😆 .

Ooo another thing I realized that's going to be a PITA for a consumer is CustomNamespace configs use the CloudWatch metric name while a supported service uses prom metric names. Perhaps we might want to move to the CloudWatch metric names for both. We know we also want to start persisting the CloudWatch name vs the prom name internally as well.

But all of this is to say we should probably split https://github.com/grafana/terraform-provider-grafana/pull/1662/files/0807df467fe5e1dc6dc755e2df617cba7fd10c46#diff-10897d974440fb02a00222ce085ea6194ec2cc20e0c99969022cbe7a797e58d4R89 in to two fields. We could call the one `CustomNamespaces` and the other `Services`, wdyt?

Reply:
The metric names I'm not 100% sure, do you think it would be confusing at all that the input to our TF provider is different than what we have listed in [our docs (the Prom names)](https://grafana.com/docs/grafana-cloud/monitor-infrastructure/aws/cloudwatch-metrics/services/)? I suppose all of the CW metric names are listed in docs as well, so it should be okay. Our Prom metric names duplicate the service names anyways, so could be less annoying to type up the CW names. I can update the field descriptions to mention that our API expects the CW metric names.

The split makes sense to me, I can work that

Reply:
If I'm honest I'm not sure why it was chosen to use expose the prom metric names. I think at one point it was helpful for a customer to see it so they could know what to look for but now we have the explore query links instead.

We can update our docs to publish both names so they are both readily available in our docs. If we use the CloudWatch names with terraform then customers who are looking at the CloudWatch docs for supported metrics aren't DOA or having to do a conversion on the fly themselves.

Reply:
Yeah I like that idea, publishing both makes sense.

Reply:
Just popping in here for future reference: docs have been updated to include both